api/pipeline_spec.proto

```diff
@@ -0,0 +1,332 @@
+syntax = "proto3";
+
+package kfp.api;
+
+import "google/protobuf/any.proto";
+
+// The spec of a pipeline.
+message PipelineSpec {
+  // The metadata of the pipeline.
+  PipelineInfo pipeline_info = 1;
+
+  // A list of pipeline tasks, which form a DAG.
+  repeated PipelineTaskSpec tasks = 2;
+
+  // The deployment config of the pipeline.
+  // The deployment config can be extended to provide platform specific configs.
+  // The supported config is [PipelineDeploymentConifg]().
+  google.protobuf.Any deployment_config = 3;
+
+  // The version of the sdk, which compiles the spec.
+  string sdk_version = 4;
+
+  // The version of the schema.
+  string schema_version = 5;
+
+  // The definition of the runtime parameter.
+  message RuntimeParameter {
+    // The type of the parameter.
+    enum Type {
+      TYPE_UNSPECIFIED = 0;
+      INT = 1;
+      DOUBLE = 2;
+      STRING = 3;
+    }
+    // Required field. The type of the runtime parameter.
+    Type type = 1;
+    // Optional field. Default value of the runtime parameter. If not set and
+    // the runtime parameter value is not provided during runtime, an error will
+    // be raised.
+    Value default_value = 2;
+  }
+  // The runtime parameters of the pipeline. Keyed by parameter name.
+  map<string, RuntimeParameter> runtime_parameters = 6;
+}

```

Comment:
+1 on this. Similar to Tekton PipelineSpec (minus k8s spec)

Reply:
Thanks, we haven't added k8s spec in this CL. This will be added later when KFP v2 started to support this IR.

@rmgogogo